rust-runtime/aws-smithy-client/src/timeout.rs

```diff
@@ -51,8 +59,257 @@ impl Settings {
     /// Sets the read timeout
     pub fn with_read_timeout(self, read_timeout: Duration) -> Self {
         Self {
-            http_read_timeout: Some(read_timeout),
+            read_timeout: Some(read_timeout),
             ..self
         }
     }
 }
+
+#[derive(Clone, Debug)]
+/// A struct containing everything needed to create a new [`TimeoutService`]
+pub struct TimeoutServiceParams {
+    /// The duration of timeouts created from these params
+    duration: Duration,
+    /// The kind of timeouts created from these params
+    kind: &'static str,
+    /// The AsyncSleep impl that will be used to create time-limited futures
+    async_sleep: Arc<dyn AsyncSleep>,
+}
+
+#[derive(Clone, Debug, Default)]
+/// A struct of structs containing everything needed to create new [`TimeoutService`]s
+pub struct AllTimeoutServiceParams {
```

Comment:
a little confused by the naming here. What about `TimeoutParams` and `ClientTimeoutParams`? Or if that seems confusing, a note about it in the docs would be useful